erpnext/manufacturing/doctype/job_card/job_card.py

```diff
@@ -522,23 +526,50 @@ def get_current_operation_data(self):
 			},
 		)
 
-	def set_transferred_qty_in_job_card(self, ste_doc):
+	def set_transferred_qty_in_job_card_item(self, ste_doc):
+		from frappe.query_builder.functions import Sum
+
+		def _validate_over_transfer(row, transferred_qty):
+			"Block over transfer of items if not allowed in settings."
+			allow_excess = frappe.db.get_single_value("Manufacturing Settings", "job_card_excess_transfer")
```

Comment:
Move this out to the outer function? So we don't need to do this computation at all if the excess transfer is allowed?

Rest LGTM.